File: app/src/main/java/alexis/tvrs/rickandmortyuniverse/ui/characters/CharacterFullActivity.java
package alexis.tvrs.rickandmortyuniverse.ui.characters;

import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.squareup.picasso.Picasso;

import alexis.tvrs.rickandmortyuniverse.R;

public class CharacterFullActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.character_full_layout);

        final Character character = (Character) getIntent().getSerializableExtra("CHARACTER");

        ImageView imageView = findViewById(R.id.characterLayout_image);
        Picasso.get().load(character.getImageUri()).into(imageView);

        TextView nameView = findViewById(R.id.characterLayout_name);
        String nameText = "Name: "+character.getName();
        nameView.setText(nameText);

        TextView statusView = findViewById(R.id.characterLayout_status);
        String statusText = "Status: "+character.getStatus();
        statusView.setText(statusText);

        TextView genderView = findViewById(R.id.characterLayout_genre);
        String genderText = "Gender: "+character.getGender();
        genderView.setText(genderText);

        TextView speciesView = findViewById(R.id.characterLayout_species);
        String speciesText = "Species: "+character.getSpecies();
        speciesView.setText(speciesText);

        final TextView originView = findViewById(R.id.characterLayout_origin);
        String originText = "Origin: "+character.getOrigin();
        originView.setText(originText);

        originView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(CharacterFullActivity.this, character.getOriginUri(), Toast.LENGTH_LONG).show();
            }
        });

        TextView locationView = findViewById(R.id.characterLayout_lastLocation);
        String lastLocationText = "Last Location: "+character.getLastLocation();
        locationView.setText(lastLocationText);

    }
}

File: app/src/main/java/alexis/tvrs/rickandmortyuniverse/ui/characters/CharacterFullActivity.kt
package alexis.tvrs.rickandmortyuniverse.ui.characters

import alexis.tvrs.rickandmortyuniverse.R
import android.os.Bundle
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.squareup.picasso.Picasso

class CharacterFullActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.character_full_layout)
        val character = intent.getSerializableExtra("CHARACTER") as Character
        val imageView = findViewById<ImageView>(R.id.characterLayout_image)
        Picasso.get().load(character.imageUri).into(imageView)
        val nameView = findViewById<TextView>(R.id.characterLayout_name)
        val nameText = "Name: " + character.name
        nameView.text = nameText
        val statusView = findViewById<TextView>(R.id.characterLayout_status)
        val statusText = "Status: " + character.status
        statusView.text = statusText
        val genderView = findViewById<TextView>(R.id.characterLayout_genre)
        val genderText = "Gender: " + character.gender
        genderView.text = genderText
        val speciesView = findViewById<TextView>(R.id.characterLayout_species)
        val speciesText = "Species: " + character.species
        speciesView.text = speciesText
        val originView = findViewById<TextView>(R.id.characterLayout_origin)
        val originText = "Origin: " + character.origin
        originView.text = originText
        originView.setOnClickListener { Toast.makeText(this, character.originUri, Toast.LENGTH_LONG).show() }
        val locationView = findViewById<TextView>(R.id.characterLayout_lastLocation)
        val lastLocationText = "Last Location: " + character.lastLocation
        locationView.text = lastLocationText
    }
}